src/components/XAxis.tsx

```diff
@@ -0,0 +1,60 @@
+import { axisBottom } from 'd3-axis';
+import { select } from 'd3-selection';
+import React, { useLayoutEffect, useRef } from 'react';
+
+import { usePlotContext } from '../hooks';
+import type { AxisProps } from '../types';
+
+const XAxis = ({ label, fontSize = 16, showTicks }: AxisProps) => {
+  const axisRef = useRef(null);
+  const { xScale, margin, height, width } = usePlotContext();
+
+  useLayoutEffect(() => {
+    if (axisRef?.current && xScale) {
+      const axis = axisBottom(xScale);
+      if (showTicks) {
+        axis.tickSizeInner((margin?.bottom || 0) + (margin?.top || 0) - height);
+      }
+
+      select(axisRef.current)
+        .call(axis)
+        .call((g) => {
+          g.selectAll('.tick line')
+            .attr('stroke-opacity', showTicks ? 0.5 : 1)
+            .attr('stroke-dasharray', showTicks ? '2,2' : '0');
+          g.selectAll('.tick text').attr(
+            'transform',
+            `translate(0,${showTicks ? 6 : 0})`,
+          );
+        });
+    }
+  });
```

Comment:
I suggest to add an array of dependencies with the values that are captured by the callback. Doesn't the linter complain about it missing? Anyway, to avoid running this code on every render, we need the array.